### PR TITLE
Layout stories

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
This adds stories for all the various combinations of `DesignType` (Article, Comment, Feature, etc) and `Display` (`showcase | immersive | standard`).

## How does it look? 👀
https://www.chromaticqa.com/library?appId=5dfcbf3012392c0020e7140b&buildNumber=393&groupPrefix=Layouts#BuildComponentListHeaderForScrollAnchor

## What's pending
My next PR will be removing the `GuardianLabs` and `Immersive` design types because they are unused and redundant, respectively.

## Why?
What if we wanted to have an immersive comment piece? This PR adds a baseline for such a combination and sets the framework for such changes in future.

## Link to supporting Trello card
https://trello.com/c/mW976Mkc/1147-article-story-coverage

### CODE REVIEW

File: src/web/layouts/DecideLayout.tsx [hunk 1]
```diff
@@ -13,7 +12,7 @@ type Props = {
 };
 
 export const DecideLayout = ({ designType, CAPI, NAV }: Props) => {
-    if (hasShowcase(CAPI.mainMediaElements)) {
+    if (CAPI.pageType.hasShowcaseMainElement) {
```
Comment (JamieB-gu) [inline src/web/layouts/DecideLayout.tsx]: Would you have a similar check here for immersive `displayHint`s once you've removed "Immersive" from `DesignType`?

Comment (oliverlloyd) [reply src/web/layouts/DecideLayout.tsx]: Yes, exactly. Once we've got an `ImmersiveLayout` built we can start using it like that. The storybook story that this ticket adds is essentially visual TDD.

**Comment by gtrufitt** [comment]
Love it. Already catches a problem with interview, showcase at wide 😉 
image

**Comment by SiAdcock** [comment]
This is one my favourite PRs of all time! ❤️ 😍 ❤️ 😍 ❤️ 😍  ❤️ 😍 

🎆 ⭐️  🎉 Super work, @oliverlloyd and the DCR team!  🎉 ⭐️ 🎆

**Comment by oliverlloyd** [comment]
So I've accepted all the stories as they are, even those with obvious problems. Mainly because we don't want this PR to turn into a catch all for our entire backlog of parity problems. This means we'll have snapshots of known issues but we can progressively fix/review them over time.

**Pull request merged**